### ISSUE TITLE
Fetching updates stalled?

### ISSUE THREAD

**Issue by E-VANCE** [outlandishideas/wpackagist#350]
Hi there,

I recently noticed that a couple of plugins weren't showing the current versions and I had to manually refresh them via https://wpackagist.org.

Not knowing which frequency you have in place for checking / fetching the plugins on the official WordPress-repo but they seem to have stalled quite a bit...?

See [this exemplary search](https://wpackagist.org/search?q=newsletter&page=2) where almost all `last fetched` entries date back to the 25th of September (I updated the first entry for the Newsletter-plugin manually).

Could you check if things are running normally?

Many thanks for providing this invaluable resource for Composer packages!

Thanks & regards

**Comment by allen-mcnichols** [comment]
Noticing the same thing, however, after manually refreshing them they seem to update as expected afterward. Might just be a matter of needing all packages to refresh in order to queue up correctly.

**Comment by grappler** [comment]
I have noticed the same thing.

**Comment by taupecat** [comment]
I've noticed this going on for at least a few weeks. And a new ticket was recently opened on this issues list that's related (#352). This is causing a disruption in my workflow as I manage over two dozen sites for clients and run a script every night to update the plugins via composer. If a plugin isn't listed as having an update, and I have to come in and manually refresh, it A) costs me time to have to do so and B) means my clients' sites aren't being updated as soon as they should be.

What can be done to fix this? Is it a matter of computing resources? People resources? Financial resources?

**Comment by khromov** [comment]
@taupecat I think it's mostly a problem with allocating enough people. I hope @NoelLH has time to look this over soon, but please keep in mind this is a free, open-source project.

You could also try using this mirror I've set up that is still using the old code/setup until the issues have been fixed in the official repo. You only need to add a couple of rows in `composer.json` to switch.
https://wpackagist.wpsv.se/

**Comment by taupecat** [comment]
That was kind of my point. If it was a problem with resources—computer, people, finance—then perhaps I could help or contribute in some way. This is a very valuable tool for my business, and if chipping in $X per year helps to keep it running smoothly, that's a worthy investment for me.

I might try your mirror, or I might use the code here to stand up my own. Either way, I thank you for your feedback.

**Comment by khromov** [comment]
@taupecat It's not that hard to set up a local mirror! We even have a [a docker-compose file](https://github.com/outlandishideas/wpackagist/blob/develop/docker-compose.yml).

**Comment by NoelLH** [comment]
Hey all, I'm aware there are longer delays than with the old approach now, and I think the full update is scheduled daily at the moment. This is not a permanent state of affairs but was necessary to get things stable until we have more time to address the slowness of the EFS filesystem and pre-existing issues that this has exacerbated.

What's happened is that the code urgently needed updates to keep its dependent libraries secure – I don't want to risk a package manager becoming a handy attack vector so this was my first priority – and the existing infrastructure tooling needed updating or replacing to get those updates out. This is how we've ended up with things on new infra before it was tested as much as I'd like.

Updating the infra approach should make the service more stable and secure in the long term. I know that's not very helpful while it is currently less stable and slower than before but I promise there is a plan! We have had to prioritise client work but I think everyone at Outlandish is keen for this to be stable and a solid resource for the WordPress community however we get there. We've now got time allocated within Outlandish to do wider pairing on this so that more of us can pick up issues including with the infrastructure, and a little extra of my time to try to figure out a better approach for updating individual packages more promptly using the refresh buttons.

There are several unknowns but we have time booked late next week, and all being well I hope the service will be running better in the live environment by mid November.

**Comment by allen-mcnichols** [comment]
So, I'm noticing that it is noticing that a plugin has been updated but the fetching process isn't running.

![image](https://user-images.githubusercontent.com/43300142/99424586-b756a580-28b6-11eb-97c3-fef7e29bac2c.png)

Manually clicking the refresh button does cause it to fetch the latest update but this can take several hours before it becomes available to update via composer.

![image](https://user-images.githubusercontent.com/43300142/99424700-d9502800-28b6-11eb-9adb-2b833f704228.png)

Do we know when this issue will be fixed or what the timeline currently looks like?

**Comment by NoelLH** [comment]
Hey all, sorry for the delay again.

We made a lot of progress in November which then stalled with one frustrating bug blocking deployment. Lots more detail is now on the specific issue https://github.com/outlandishideas/wpackagist/issues/368

PRs to fix that are very welcome and would unblock testing all the latest improvements on Staging.

**Comment by NoelLH** [comment]
Staging update command has run and this looks to be in place now without a manual update. I checked the version mentioned on #367 which was missing before (Staging was previously set to only update once every 30 days) and it's there now.

**Comment by NoelLH** [comment]
That partial update took 5m 12s. So while the difference in resources was not very scientific (I don't know how much of the time difference was down to the update being smaller) this seems like a pretty reasonable level to run the task at hourly in Production without costs spiralling and with a low risk of task invocations overlapping.

I'll reduce Staging's frequency to daily now and set up Production with the hourly config I just tested.

**Comment by NoelLH** [comment]
Prod code is out. It's running an update I invoked manually so update commands will be flaky for a few more mins before hopefully stabilising long-term.

**Comment by NoelLH** [comment]
Changes all live and scheduled updates are now running hourly.

Improved on-demand updates are also out, so you can fully refresh a specific package's versions without waiting for the next full update.
